File: 2509-minimize-xor/2509-minimize-xor.py
class Solution:
    def minimizeXor(self, num1: int, num2: int) -> int:
        bc = bin(num2).count('1')
        bc -= bin(num1).count('1')
        cur = 1
        while bc != 0:
            if bc < 0 and (num1 & cur) != 0:
                num1 ^= cur
                bc += 1
            elif bc > 0 and (num1 & cur) == 0:
                num1 |= cur
                bc -= 1
            cur <<= 1
        return num1

File: 2509-minimize-xor/2509-minimize-xor.java
class Solution {
    public int minimizeXor(int num1, int num2) {
        int bc = Integer.bitCount(num2);
        bc -= Integer.bitCount(num1);

        int cur = 1;
        while (bc != 0) {
            if (bc < 0 && (num1 & cur) != 0) {
                num1 ^= cur;
                bc++;
            }
            else if (bc > 0 && (num1 & cur) == 0) {
                num1 |= cur;
                bc--;
            }
            cur <<= 1;
        }
        return num1;
    }
}